pkg/gateway/server/user.go

```diff
@@ -206,15 +201,18 @@ func (s *Server) deleteUser(apiContext api.Context) (err error) {
 		return fmt.Errorf("failed to start deletion of user owned objects: %v", err)
 	}
 
-	// Tell the browser to remove the access token cookie, so that the user does not immediately attempt to authenticate again.
-	http.SetCookie(apiContext.ResponseWriter, &http.Cookie{
-		Name:     proxy.ObotAccessTokenCookie,
-		Value:    "",
-		Path:     "/",
-		MaxAge:   -1,
-		HttpOnly: true,
-		Secure:   strings.HasPrefix(s.uiURL, "https://"),
-	})
+	// Only clear the cookie if this is a "delete me" operation
+	if isDeleteMe {
```

Comment:
Sorry about this! It was my fault that we were deleting the token for everyone, and not just "delete me" requests. I didn't consider that at first when I added this code.